examples/lines_bars_and_markers/timeline.py

```diff
@@ -15,57 +15,98 @@
 import matplotlib.dates as mdates
 from datetime import datetime
 
-# A list of Matplotlib releases and their dates
-# Taken from https://api.github.com/repos/matplotlib/matplotlib/releases
-names = ['v2.2.2', 'v2.2.1', 'v2.2.0', 'v2.1.2', 'v2.1.1', 'v2.1.0', 'v2.0.2',
-         'v2.0.1', 'v2.0.0', 'v1.5.3', 'v1.5.2', 'v1.5.1', 'v1.5.0', 'v1.4.3',
-         'v1.4.2', 'v1.4.1', 'v1.4.0']
-
-dates = ['2018-03-17T03:00:07Z', '2018-03-16T22:06:39Z',
-         '2018-03-06T12:53:32Z', '2018-01-18T04:56:47Z',
-         '2017-12-10T04:47:38Z', '2017-10-07T22:35:12Z',
-         '2017-05-10T02:11:15Z', '2017-05-02T01:59:49Z',
-         '2017-01-17T02:59:36Z', '2016-09-09T03:00:52Z',
-         '2016-07-03T15:52:01Z', '2016-01-10T22:38:50Z',
-         '2015-10-29T21:40:23Z', '2015-02-16T04:22:54Z',
-         '2014-10-26T03:24:13Z', '2014-10-18T18:56:23Z',
-         '2014-08-26T21:06:04Z']
-dates = [datetime.strptime(ii, "%Y-%m-%dT%H:%M:%SZ") for ii in dates]
+try:
+    # Try to fetch a list of Matplotlib releases and their dates
+    # from https://api.github.com/repos/matplotlib/matplotlib/releases
+    import urllib.request
+    import json
+
+    url = 'https://api.github.com/repos/matplotlib/matplotlib/releases'
+    url += '?per_page=100'
+    data = json.loads(urllib.request.urlopen(url, timeout=.4).read().decode())
+
+    dates = []
+    names = []
+    for item in data:
+        if 'rc' not in item['tag_name'] and 'b' not in item['tag_name']:
+            dates.append(item['published_at'].split("T")[0])
+            names.append(item['tag_name'])
+except:
```

Comment:
```suggestion
except Exception:
```

No bare except (see last paragraph of https://docs.python.org/2/howto/doanddont.html#except). Not that it really matters here, but we shouldn't use bad style.